pkg/gcv/configs/config.go

```diff
@@ -380,19 +389,58 @@ func (c *Configuration) loadUnstructured(u *unstructured.Unstructured) error {
 			return errors.Wrapf(err, "failed to convert to versioned constraint template internal struct")
 		}
 
-		c.Templates = append(c.Templates, &ct)
-	case u.GroupVersionKind().Group == constraintGroup:
-		if err := convertLegacyConstraint(u); err != nil {
-			return errors.Wrapf(err, "failed to convert constraint")
+		for _, target := range ct.Spec.Targets {
+			switch target.Target {
+			// TODO: Using consant from gcptarget package causes circular reference.  Fix circular reference and use gcptarget.Name
+			case "validation.gcp.forsetisecurity.org":
+				c.GCPTemplates = append(c.GCPTemplates, &ct)
+			case "admission.k8s.gatekeeper.sh":
+				c.K8STemplates = append(c.K8STemplates, &ct)
+			default:
+				return errors.Errorf("")
+			}
 		}
 
-		c.Constraints = append(c.Constraints, u)
+	case u.GroupVersionKind().Group == constraintGroup:
+		c.allConstraints = append(c.allConstraints, u)
+
 	default:
 		return errors.Errorf("unexpected data type %s", u.GroupVersionKind())
 	}
 	return nil
 }
 
+func (c *Configuration) finishLoad() error {
+	templates := map[string]string{}
+	for _, t := range c.GCPTemplates {
+		templates[t.Spec.CRD.Spec.Names.Kind] = gcpConstraint
+	}
+	for _, t := range c.K8STemplates {
+		templates[t.Spec.CRD.Spec.Names.Kind] = k8sConstraint
+	}
+
+	allConstraints := c.allConstraints
+	c.allConstraints = nil
```

Comment:
Looks like allConstraints is only used in NewConfiguration()?
In that case, does it need to be in Configuration? Should we move it to be inside NewConfiguration() instead? Otherwise it's odd to see an unused field in the struct...

Reply:
could you add a comment around line 322 to mention this?

Reply:
added